lib/rubocop/cop/rspec/class_check.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Enforces consistent use of `be_a` or `be_kind_of`.
+      #
+      # @example EnforcedStyle: be_a (default)
+      #   # bad
+      #   expect(object).to be_kind_of(String)
+      #   expect(object).to be_a_kind_of(String)
+      #
+      #   # good
+      #   expect(object).to be_a(String)
+      #   expect(object).to be_an(String)
+      #
+      # @example EnforcedStyle: be_kind_of
+      #   # bad
+      #   expect(object).to be_a(String)
+      #   expect(object).to be_an(String)
+      #
+      #   # good
+      #   expect(object).to be_kind_of(String)
+      #   expect(object).to be_a_kind_of(String)
+      #
+      class ClassCheck < Base
+        extend AutoCorrector
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Prefer `%<preferred>s` over `%<current>s`.'
+
+        METHOD_NAMES_FOR_BE_A = ::Set[
+          :be_a,
+          :be_an
+        ].freeze
+
+        METHOD_NAMES_FOR_KIND_OF = ::Set[
+          :be_a_kind_of,
+          :be_kind_of
+        ].freeze
+
+        RESTRICT_ON_SEND = %i[
+          be_a
+          be_a_kind_of
+          be_an
+          be_kind_of
+        ].freeze
+
+        def on_send(node)
+          return if preferred_method_name?(node.method_name)
+
+          add_offense(
+            node.loc.selector,
+            message: format_message(node)
+          ) do |corrector|
+            autocorrect(corrector, node)
+          end
+        end
+
+        private
+
+        def autocorrect(corrector, node)
+          corrector.replace(node.loc.selector, style)
+        end
+
+        def format_message(node)
+          format(
+            MSG,
+            current: node.method_name,
+            preferred: style
+          )
+        end
```

Comment:
I think I am [directly contradicting](https://github.com/rubocop/rubocop-rspec/pull/1357#discussion_r938711003) @ydah when I say that I don’t like using `style` in line 63 and 70. The name of the style and the preferred literal code are _incidentally_ identical, but they are two different concepts. While it’s not a big deal here (we probably won’t change the style names soon) it just doesn’t look right to me.

⬆️ This is just my opinion, not necessarily the truth 😄

Reply:
Certainly, you have a point.

One of the reasons I chose to use `style` here is that it is better to calculate preferred method names based on `style`, rather than based on `node`. I think we can probably agree on this point. On top of that, if we want to make it explicit that style and method name are not the same concept, we should put a map from style to preferred method name, even if it seems a bit redundant.

Reply:
I agree that the map would be the best approach. And having it map from style to preferred code (e.g. `{ be_a: 'be_a', be_kind_of: 'be_kind_of' }`) does seem a bit redundant.